.circleci/utils.py

```diff
@@ -0,0 +1,8 @@
+import urllib, json, sys, subprocess
+
+def build_and_push_docker(dir, container, tag):
+  build_command = ['docker', 'build', '-t', '%s:%s' % (container, tag), dir]
+  push_command = ['docker', 'push', '%s:%s' % (container, tag)]
+  build_code = subprocess.call(build_command)
```

Comment:
Rather than using `subprocess.call`, you might want to look at [`subprocess.check_call`](https://docs.python.org/3/library/subprocess.html?highlight=check_call#subprocess.check_call) – same as `call()`, but it throws an exception if it exits with a non-zero exit code. Saves you having to do the bookkeeping yourself.